app/src/main/java/by/hometrainng/listapi/fragments/CharacterFragment.kt

```diff
@@ -0,0 +1,97 @@
+import by.hometrainng.listapi.retrofit.FinalSpaceService
+import coil.load
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.HttpException
+import retrofit2.Response
+
+class CharacterFragment: Fragment() {
+
+    private var _binding: FragmentCharacterBinding? = null
+    private val binding get() = requireNotNull(_binding)
+    private val args by navArgs<CharacterFragmentArgs>()
+    private var currentCall: Call<ListElement.Character>? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return FragmentCharacterBinding.inflate(inflater, container, false)
+            .also { _binding = it }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        with (binding) {
+            val characterId = args.characterId.toString()
+            toolbarCharacter.setupWithNavController(findNavController())
+            loadCharacterDetails(characterId, binding)
+        }
+    }
+
+    private fun loadCharacterDetails (id: String, binding: FragmentCharacterBinding) {
+        if (currentCall == null) {
+            currentCall = FinalSpaceService.provideFinalSpaceApi().getCharacterDetails(id)
+            currentCall?.enqueue(object : Callback<ListElement.Character> {
+                @SuppressLint("SetTextI18n")
+                override fun onResponse(
+                    call: Call<ListElement.Character>,
+                    response: Response<ListElement.Character>
+                ) {
+                    if (response.isSuccessful) {
+                        val character = response.body() ?: return
+                        with(binding) {
+                            image.load(character.imageURL)
+                            name.text = character.name
+                            species.text = SPECIES + character.species
+                            status.text = STATUS + character.status
+                            gender.text = GENDER + character.gender
+                            hair.text = HAIR + character.hair
+                        }
+                    } else {
+                        showToastMessage(HttpException(response).message())
+                    }
+                    currentCall = null
+                }
+                override fun onFailure(call: Call<ListElement.Character>, t: Throwable) {
+                    currentCall = null
+                    showToastMessage(UPLOAD_FAILURE)
+                }
+            })
+        }
+    }
+
+    private fun showToastMessage(message: String) {
+        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    companion object {
+        const val SPECIES = "Species: "
```

Comment:
константы лучше приватными делать, также можно воспользоваться шаблонами
```suggestion
        const val SPECIES = "Species: %s"
```
и для форматирования тогда будет `SPECIES.format(character.species)`